mne/decoding/transformer.py

```diff
@@ -206,43 +207,70 @@ def transform(self, epochs_data, y=None):
         -------
         X : array, shape (n_epochs, n_channels * n_times)
             The data concatenated over channels
+        y : array, shape(n_epochs,)
+            The label for each epoch.
         """
-        if not isinstance(epochs_data, np.ndarray):
-            raise ValueError("epochs_data should be of type ndarray (got %s)."
-                             % type(epochs_data))
+        if not isinstance(epochs, _BaseEpochs):
+            raise ValueError("epochs should be an instance of Epochs got %s"
+                             " instead" % type(epochs))
 
-        epochs_data = np.atleast_3d(epochs_data)
+        self.epochs_data = epochs._data
```

Comment:
Don't store the epochs data in self! The transformer object must remain as light as possible